packages/contracts/tasks/take-dump.ts

```diff
@@ -1,83 +1,70 @@
-  // Address of the L1CrossDomainMessenger contract.
-  l1CrossDomainMessengerAddress: string
-  // Block height to activate berlin hardfork
-  berlinBlock: number
-}
-
-/**
- * Generates the initial state for the L2 system by injecting the relevant L2 system contracts.
- *
- * @param cfg Configuration for the L2 system.
- * @returns Generated L2 genesis state.
- */
-export const makeL2GenesisFile = async (
-  cfg: RollupDeployConfig
-): Promise<any> => {
-  // Very basic validation.
-  for (const [key, val] of Object.entries(cfg)) {
-    if (val === undefined) {
-      throw new Error(`must provide an input for config value: ${key}`)
-    }
+import { predeploys } from '../src/predeploys'
+import { getDeployConfig, getContractFromArtifact } from '../src/deploy-utils'
+import { names } from '../src/address-names'
+
+task('take-dump').setAction(async (args, hre) => {
+  /* eslint-disable @typescript-eslint/no-var-requires */
+
+  // Needs to be imported here or hardhat will throw a fit about hardhat being imported from
+  // within the configuration file.
+  const {
+    computeStorageSlots,
+    getStorageLayout,
+  } = require('@defi-wonderland/smock/dist/src/utils')
+
+  // Needs to be imported here because the artifacts can only be generated after the contracts have
+  // been compiled, but compiling the contracts will import the config file which, as a result,
+  // will import this file.
+  const { getContractArtifact } = require('../src/contract-artifacts')
+
+  /* eslint-enable @typescript-eslint/no-var-requires */
+
+  // Make sure we have a deploy config for this network
+  const deployConfig = getDeployConfig(hre.network.name)
+
+  // Basic warning so users know that the whitelist will be disabled if the owner is the zero address.
+  if (
+    deployConfig.ovmWhitelistOwner === undefined ||
+    deployConfig.ovmWhitelistOwner === ethers.constants.AddressZero
+  ) {
+    console.log(
+      'WARNING: whitelist owner is undefined or address(0), whitelist will be disabled'
+    )
   }
 
   const variables = {
     OVM_DeployerWhitelist: {
-      owner: cfg.whitelistOwner,
+      owner: deployConfig.ovmWhitelistOwner || ethers.constants.AddressZero,
     },
     OVM_GasPriceOracle: {
-      _owner: cfg.gasPriceOracleOwner,
-      gasPrice: cfg.gasPriceOracleGasPrice,
-      l1BaseFee: cfg.gasPriceOracleL1BaseFee,
-      overhead: cfg.gasPriceOracleOverhead,
-      scalar: cfg.gasPriceOracleScalar,
-      decimals: cfg.gasPriceOracleDecimals,
+      _owner: deployConfig.ovmGasPriceOracleOwner,
+      gasPrice: deployConfig.gasPriceOracleL2GasPrice || 1,
+      l1BaseFee: deployConfig.gasPriceOracleL1BaseFee || 1,
+      overhead: deployConfig.gasPriceOracleOverhead || 2750,
+      scalar: deployConfig.gasPriceOracleScalar || 1_500_000,
+      decimals: deployConfig.gasPriceOracleDecimals || 6,
     },
     L2StandardBridge: {
-      l1TokenBridge: cfg.l1StandardBridgeAddress,
+      l1TokenBridge: (
+        await getContractFromArtifact(
```

Comment:
This is no longer configurable?

Reply:
Correct, it gets pulled automatically from the deployment folder based on the current network name. This is effectively what we did anyway, except we did it manually which is error prone.